geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ConvertCommand.scala

```diff
@@ -0,0 +1,110 @@
+      else throw new ParameterException("Unable to parse Simple Feature type from sft config or string")
+    }
+    val converter = SimpleFeatureConverters.build(sft, converterConfig)
+
+    val outFmt = DataFormats.values.find(_.toString.equalsIgnoreCase(params.outputFormat)).getOrElse {
+      throw new ParameterException("Unable to parse output file type.")
+    }
+
+    lazy val outputStream: OutputStream = ExportCommand.createOutputStream(params.file, params.gzip)
+    val writer: Writer = ExportCommand.getWriter(params.asInstanceOf[ExportParams])
+
+    val exporter = outFmt match {
+      case Csv | Tsv      => new DelimitedExporter(writer, outFmt, !params.noHeader)
+      case Shp            =>
+        Seq(ShapefileExporter.modifySchema(sft))
+        new ShapefileExporter(ExportCommand.checkShpFile(params.asInstanceOf[ExportParams]))
+      case GeoJson | Json => new GeoJsonExporter(writer)
+      case Gml            => new GmlExporter(outputStream)
+      case Avro           =>
+        val avroCompression = Option(params.gzip).map(_.toInt).getOrElse(Deflater.DEFAULT_COMPRESSION)
+        new AvroExporter(sft, outputStream, avroCompression)
+      case Bin            =>
+        if(Seq(params.idAttribute, params.latAttribute, params.lonAttribute, params.labelAttribute).contains(null))
+          throw new ParameterException("Missing parameters for binary export. "
+            + "For more information use: ./geomesa convert --help")
+        BinExporter(outputStream, params.asInstanceOf[BinExportParams], sft.getDtgField)
+      case _              => throw new ParameterException(s"Format $outFmt is not supported.")
+    }
+
+    val filter = Option(params.cqlFilter).map(ECQL.toFilter).getOrElse(Filter.INCLUDE)
+    val fc = new DefaultFeatureCollection(sft.getTypeName, sft)
+
+    try {
+      params.files.foreach{ file =>
+        val ec = converter.createEvaluationContext(Map("inputFilePath" -> file))
+        val dataIter = converter.process(new FileInputStream(file.toString), ec)
+        if (params.maxFeatures != null && params.maxFeatures >= 0) {
+          logger.info(s"Converting ${getPlural(params.maxFeatures.toLong, "feature")} from $file")
+          for (i <- 1 to params.maxFeatures) {
+            if (dataIter.hasNext) fc.add(dataIter.next())
+          }
+        } else dataIter.foreach(fc.add)
+        logger.info(s"Converted ${getPlural(ec.counter.getLineCount, "simple feature")} "
+          + s"with ${getPlural(ec.counter.getSuccess, "success", "successes")} "
+          + s"and ${getPlural(ec.counter.getFailure, "failure")}")
+      }
+      logger.debug(s"Applying CQL filter ${filter.toString}")
+      exporter.export(fc.subCollection(filter))
+    } finally {
+      IOUtils.closeQuietly(converter)
+      IOUtils.closeQuietly(exporter)
+    }
+  }
+}
+
+object ConvertParameters {
+  @Parameters(commandDescription = "Convert files using GeoMesa's internal SFT converter framework")
```

Comment:
[picky] I would remove 'SFT' here and just refer to 'internal converter framework'